server/src/resources/parcel_scan/mod.rs

```diff
@@ -0,0 +1,145 @@
+//! Parcel Scan
+
+pub use crate::grpc::server::parcel_scan::*;
+
+use log::debug;
+use std::collections::HashMap;
+use tokio_postgres::row::Row;
+use tokio_postgres::types::Type as PsqlFieldType;
+use uuid::Uuid;
+
+use super::base::simple_resource::*;
+use super::base::{FieldDefinition, ResourceDefinition};
+use crate::common::ArrErr;
+use crate::grpc::{GrpcDataObjectType, GrpcField};
+use crate::resources::GeoPoint;
+
+crate::build_generic_resource_impl_from!();
+
+// Generate grpc server implementations
+crate::build_grpc_simple_resource_impl!(parcel_scan);
+
+impl Resource for ResourceObject<Data> {
+    fn get_definition() -> ResourceDefinition {
+        ResourceDefinition {
+            psql_table: String::from("parcel_scan"),
+            psql_id_cols: vec![String::from("parcel_scan_id"), String::from("created_at")],
+            fields: HashMap::from([
+                (
+                    "parcel_id".to_string(),
+                    FieldDefinition::new(PsqlFieldType::UUID, true),
+                ),
+                (
+                    "scanner_id".to_string(),
+                    FieldDefinition::new(PsqlFieldType::UUID, true),
+                ),
+                (
+                    "geo_location".to_string(),
+                    FieldDefinition::new(PsqlFieldType::POINT, true),
+                ),
+                (
+                    "created_at".to_string(),
+                    FieldDefinition::new_internal(PsqlFieldType::TIMESTAMPTZ, true)
+                        .set_default(String::from("CURRENT_TIMESTAMP")),
+                ),
+                (
+                    "updated_at".to_string(),
+                    FieldDefinition::new_internal(PsqlFieldType::TIMESTAMPTZ, true)
+                        .set_default(String::from("CURRENT_TIMESTAMP")),
+                ),
+                (
+                    "deleted_at".to_string(),
+                    FieldDefinition::new_internal(PsqlFieldType::TIMESTAMPTZ, true)
+                        .set_default(String::from("CURRENT_TIMESTAMP")),
+                ),
+            ]),
+        }
+    }
+
+    fn get_table_indices() -> Vec<String> {
+        [
+            r#"ALTER TABLE parcel_scan ADD CONSTRAINT fk_parcel_id FOREIGN KEY(parcel_id) REFERENCES parcel(parcel_id)"#.to_owned(),
+            r#"ALTER TABLE parcel_scan ADD CONSTRAINT fk_scanner_id FOREIGN KEY(scanner_id) REFERENCES scanner(scanner_id)"#.to_owned(),
```

Comment:
There should also be a unique key on the parcel_id, scanner_id, geo_location combination.

Reply:
[cc969e4](https://github.com/Arrow-air/svc-storage/pull/62/commits/cc969e45bda8b46c79ae04c6c37063bdc700886e)